### PR TITLE
Controller: Add Analog Joystick support

### PR DESCRIPTION
**Pull request by ggrtk**
Allows the use of analog controls for pre-Analog Controller titles such as Descent and MechWarrior 2.

### CODE REVIEW

File: src/core/analog_controller.cpp [hunk 1]
```diff
@@ -41,8 +42,10 @@ bool AnalogController::DoState(StateWrapper& sw)
 
   sw.Do(&m_analog_mode);
   sw.Do(&m_rumble_unlocked);
+  sw.Do(&m_legacy_rumble_unlocked);
```
Comment (stenzek) [inline src/core/analog_controller.cpp:45]: This should be `sw.DoEx(&m_legacy_rumble_unlocked, 44, false);`

File: src/core/analog_controller.cpp [hunk 2]
```diff
@@ -41,8 +42,10 @@ bool AnalogController::DoState(StateWrapper& sw)
...
   sw.Do(&m_configuration_mode);
   sw.Do(&m_command_param);
+  sw.Do(&m_button_state);
```
Comment (stenzek) [inline src/core/analog_controller.cpp:48]: Same here

**Review by stenzek — COMMENTED** [review]
With the version check you don't need to put SAVEVERSION in the commit, since the old states should load fine.

**Comment by ggrtk** [comment]
Didn't know about `DoEx`, thanks.

**Pull request merged**